internal/ethapi/api.go

```diff
@@ -893,6 +902,54 @@ func (diff *StateOverride) Apply(state *state.StateDB) error {
 	return nil
 }
 
+func newRPCBalance(balance *big.Int) **hexutil.Big {
+	rpcBalance := (*hexutil.Big)(balance)
+	return &rpcBalance
+}
+
+func CalculateL1MsgFee(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *StateOverride, timeout time.Duration, globalGasCap uint64, config *params.ChainConfig) (*big.Int, error) {
+	if !config.UsingScroll {
+		return big.NewInt(0), nil
+	}
+
+	state, header, err := b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+	if state == nil || err != nil {
+		return nil, err
+	}
+	if err := overrides.Apply(state); err != nil {
```

Comment:
if `overrides` is `nil` can it still `Apply(state)`?

Reply:
Yes, for `nil` it's noop. https://github.com/scroll-tech/go-ethereum/blob/develop/internal/ethapi/api.go#L863